pulp_container/app/registry_api.py

```diff
@@ -272,7 +275,8 @@ def get_drv_pull(self, path):
         try:
             distribution = models.ContainerDistribution.objects.get(base_path=path)
         except models.ContainerDistribution.DoesNotExist:
-            raise RepositoryNotFound(name=path)
+            # get a pull-through cache distribution whose base_path is a substring of path
+            return self.get_pull_through_drv(path)
```

Comment:
Another question is whether we want to use Redis caching in the first place. I am open to enhancing handle_safe_method() to detect if we are doing pull-through caching. If not, the decorator will remain; otherwise, we will proceed to real-time pull-through caching.

Reply:
We decided to leave the caching in place. Shortly, we will remove this since TTL is a globally configurable option that might negatively impact the feature.